pallets/creators-staking/src/lib.rs

```diff
@@ -0,0 +1,878 @@
+            ensure!(staked_value > Zero::zero(), Error::<T>::NotStakedCreator);
+
+            // Calculate the value which will be unstaked.
+            let remaining = staked_value.saturating_sub(value);
+            let value_to_unstake = if remaining < T::MinimumStakingAmount::get() {
+                stake_info.number_of_stakers = stake_info.number_of_stakers.saturating_sub(1);
+                staked_value
+            } else {
+                value
+            };
+            stake_info.total = stake_info.total.saturating_sub(value_to_unstake);
+
+            // Sanity check
+            ensure!(value_to_unstake > Zero::zero(), Error::<T>::CannotUnstakeZero);
+
+            staker_info
+                .unstake(current_era, value_to_unstake)
+                .map_err(|_| Error::<T>::UnexpectedStakeInfoEra)?;
+            ensure!(
+                // One spot should remain for compounding reward claim call
+                staker_info.len() < T::MaxEraStakeValues::get(),
+                Error::<T>::TooManyEraStakeValues
+            );
+
+            Ok(value_to_unstake)
+        }
+
+        /// Update the ledger for a staker. This will also update the stash lock.
+        /// This lock will lock the entire funds except paying for further transactions.
+        fn update_ledger(staker: &T::AccountId, ledger: AccountLedger<BalanceOf<T>>) {
+            if ledger.is_empty() {
+                Ledger::<T>::remove(&staker);
+                T::Currency::remove_lock(STAKING_ID, staker);
+            } else {
+                T::Currency::set_lock(STAKING_ID, staker, ledger.locked, WithdrawReasons::all());
+                Ledger::<T>::insert(staker, ledger);
+            }
+        }
+
+        /// Update the staker info for the `(staker, creator_id)` pairing.
+        /// If staker_info is empty, remove it from the DB. Otherwise, store it.
+        fn update_staker_info(
+            staker: &T::AccountId,
+            creator_id: SpaceId,
+            staker_info: StakerInfo<BalanceOf<T>>,
+        ) {
+            if staker_info.is_empty() {
+                GeneralStakerInfo::<T>::remove(staker, creator_id)
+            } else {
+                GeneralStakerInfo::<T>::insert(staker, creator_id, staker_info)
+            }
+        }
+
+        /// Calculate reward split between a creator and its stakers.
+        ///
+        /// Returns (creator reward, joint stakers reward)
+        pub(crate) fn creator_stakers_split(
+            creator_info: &CreatorStakeInfo<BalanceOf<T>>,
+            era_info: &EraInfo<BalanceOf<T>>,
+        ) -> (BalanceOf<T>, BalanceOf<T>) {
+            let creator_stake_portion =
+                Perbill::from_rational(creator_info.total, era_info.staked);
+
+            let creator_reward_part = creator_stake_portion * era_info.rewards.creators;
+            let stakers_joint_reward = creator_stake_portion * era_info.rewards.stakers;
+
+            (creator_reward_part, stakers_joint_reward)
```

Comment:
stakers_joint_reward looks as a bad name

Reply:
I changed the `creator_stakers_split` function name to `distributed_rewards_between_creator_and_stakers` (without `get_` prefix, _read more below)_ to make it clear that it's handling the distribution of rewards. Inside the function, I renamed the variables to better describe their purpose and relationship to the rewards for the creator and the stakers.

**Important notice:**
Due to Rust API Guildines (naming convention) with a few exceptions, the `get_` prefix is not used for getters in Rust code. The `get` naming is used only when there is a single and obvious thing that could reasonably be gotten by a getter.